src/diffusers/optimization.py

```diff
@@ -77,6 +78,48 @@ def lr_lambda(current_step: int):
     return LambdaLR(optimizer, lr_lambda, last_epoch=last_epoch)
 
 
+def get_constant_schedule_with_rules(optimizer: Optimizer, rules: str, last_epoch: int = -1):
```

Comment:
```suggestion
def get_piecewise_constant_schedule(optimizer: Optimizer, steps: str, last_epoch: int = -1):
```

If we agree on the naming suggested above